core/compilePatterns.ml

```diff
@@ -13,21 +13,6 @@ open Utility
 open Ir
 open SugarConstructors.DummyPositions
 
-type pattern = [
-| `Any
-| `Nil
-| `Cons     of pattern * pattern
-| `Variant  of name * pattern
-| `Effect   of name * pattern list * pattern
-| `Negative of StringSet.t
-| `Record   of (pattern StringMap.t) * pattern option
-| `Constant of Constant.t
-| `Variable of binder
-| `As       of binder * pattern
-| `HasType  of pattern * Types.datatype
-]
-    [@@deriving show]
-
 module Const = struct
```

Comment:
Perhaps instead of having another module for representing constants we could extend existing module `CommonTypes.Constant` with `compare` function and use it instead?

Reply:
Okay, I've given CommonTypes.Constant a derived compare function. This should actually be faster than the polymorphic Pervasives.compare function.